File: leetcode/reverse_words_in_a_string.cpp
class Solution {
public:
    void reverseword(string &s, int i,int j){
        while(i<j){
            char tmp=s[i];
            s[i++]=s[j];
            s[j--]=tmp;
        }
    }
    
    string reverseWords(string s) {
        int i=0,j=0,l=0;
        int n=s.length();
        int wordCount=0;
        while(true){
            while(i<n && s[i]==' ') i++;
            if(i==n) break;
            if(wordCount) s[j++]=' ';
            l=j;
            while(i<n && s[i]!=' '){
                s[j]=s[i];
                j++;i++;
            }
            reverseword(s,l,j-1);
            wordCount++;
        }
        s.resize(j);
        reverseword(s,0,j-1);
        return s;
    }
};

File: leetcode/reverse_words_in_a_string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        s = s.strip()
        l = s.split()
        l = [i for i in l]
        l.reverse()
        s = ""
        for i in l:
            x= i.strip() 
            if (x == ""):
                continue
            else:
                s += i + " "
        return s.strip()
